Assets/Scripts/PlayerController.cs

```diff
@@ -69,9 +72,18 @@ public class PlayerController : MonoBehaviour {
         // TODO: Add "speed limit" (top speed) here
 
         // Record players last known location, using the center of the ball
-        if (Physics.Raycast(this.transform.position, Vector3.down, 0.6f)) {
+        int movingPlatformMask = 1 << 8; // This is the layer mask for moving platforms
+
+        // NOTE: All moving platforms must use this layer (use the moving platform prefab)
+
+        if (Physics.Raycast(this.transform.position, Vector3.down, 0.6f, ~movingPlatformMask)) {
             playerLastPosition = this.transform.position;
 
+            // This will remove the reference to the last platform the player was on for reset purposes
+            if (this.playerLastPlatform != null) {
```

Comment:
To get really nit picky, this if check doesn't do anything, right? Should always be able to just write null into the variable.